### PR TITLE
HBASE-24897 RegionReplicaFlushHandler should handle NoServerForRegionException to avoid aborting RegionServer

### PR DESCRIPTION
**Pull request by infraio**
…ion replica

### CODE REVIEW

File: hbase-protocol-shaded/src/main/protobuf/MasterProcedure.proto [hunk 1]
```diff
@@ -53,9 +53,10 @@ enum CreateTableState {
   CREATE_TABLE_PRE_OPERATION = 1;
   CREATE_TABLE_WRITE_FS_LAYOUT = 2;
   CREATE_TABLE_ADD_TO_META = 3;
-  CREATE_TABLE_ASSIGN_REGIONS = 4;
-  CREATE_TABLE_UPDATE_DESC_CACHE = 5;
-  CREATE_TABLE_POST_OPERATION = 6;
+  CREATE_TABLE_ASSIGN_PRIMARY_REGIONS = 4;
```
Comment (Apache9) [inline hbase-protocol-shaded/src/main/protobuf/MasterProcedure.proto:56]: This is an incompatible change? If we have a pending CreateTableProcedure when upgrading we will in trouble?

Comment (infraio) [reply hbase-protocol-shaded/src/main/protobuf/MasterProcedure.proto:56]: So all procedures proto changes are incompatible changes? The design of procedure needs to be very careful......

Comment (Apache9) [reply hbase-protocol-shaded/src/main/protobuf/MasterProcedure.proto:56]: Yes, so usually we do not change the states for a patch release...

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   2m  2s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  prototool  |   0m  0s |  prototool was not available.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ branch-2.2 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 11s |  branch-2.2 passed  |
| +1 :green_heart: |  compile  |   1m 34s |  branch-2.2 passed  |
| +1 :green_heart: |  checkstyle  |   1m 30s |  branch-2.2 passed  |
| +1 :green_heart: |  shadedjars  |   4m  4s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 52s |  branch-2.2 passed  |
| +0 :ok: |  spotbugs  |   3m 15s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   5m 41s |  branch-2.2 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 54s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 36s |  the patch passed  |
| +1 :green_heart: |  cc  |   1m 36s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 36s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 29s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m  5s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  25m  9s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 2.10.0 or 3.1.2 3.2.1.  |
| +1 :green_heart: |  hbaseprotoc  |   1m 28s |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 50s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   5m 40s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 41s |  hbase-protocol-shaded in the patch passed.  |
| +1 :green_heart: |  unit  | 160m 40s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m  8s |  The patch does not generate ASF License warnings.  |
|  |   | 231m 34s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.12 Server=19.03.12 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2271/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2271 |
| Optional Tests | dupname asflicense cc unit hbaseprotoc prototool javac javadoc spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 57f30e136875 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-home/workspace/Base-PreCommit-GitHub-PR_PR-2271/out/precommit/personality/provided.sh |
| git revision | branch-2.2 / 8329591b45 |
| Default Java | 1.8.0_181 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2271/1/testReport/ |
| Max. process+thread count | 3699 (vs. ulimit of 12500) |
| modules | C: hbase-protocol-shaded hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2271/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by huaxiangsun** [comment]
Catch the exception here? Wondering this could happen after the regions are assigned as well.

File: hbase-protocol-shaded/src/main/protobuf/MasterProcedure.proto (hunk 1 above)
Comment (infraio) [reply hbase-protocol-shaded/src/main/protobuf/MasterProcedure.proto]: Ok. Will just fix this by another way: catch the error and retry to avoid abort RS.

**Comment by infraio** [comment]
> Catch the exception here? Wondering this could happen after the regions are assigned as well.

Yes. Not the best way but useful now.

**Review by Apache9 — APPROVED** [review]
+1

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 29s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ branch-2.2 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 50s |  branch-2.2 passed  |
| +1 :green_heart: |  compile  |   1m 12s |  branch-2.2 passed  |
| +1 :green_heart: |  checkstyle  |   1m 42s |  branch-2.2 passed  |
| +1 :green_heart: |  shadedjars  |   4m 50s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 40s |  branch-2.2 passed  |
| +0 :ok: |  spotbugs  |   3m 29s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m 27s |  branch-2.2 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 54s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 55s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 55s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 15s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m  5s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  25m 59s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 2.10.0 or 3.1.2 3.2.1.  |
| +1 :green_heart: |  javadoc  |   0m 37s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   3m 37s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 176m 41s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 36s |  The patch does not generate ASF License warnings.  |
|  |   | 240m 49s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.12 Server=19.03.12 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2271/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2271 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux d2dcc65d5192 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-home/workspace/Base-PreCommit-GitHub-PR_PR-2271/out/precommit/personality/provided.sh |
| git revision | branch-2.2 / 8329591b45 |
| Default Java | 1.8.0_181 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2271/2/testReport/ |
| Max. process+thread count | 2861 (vs. ulimit of 12500) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2271/2/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**